dashboard/app/models/user.rb

```diff
@@ -881,26 +881,60 @@ def in_progress_and_completed_scripts
     end
   end
 
-  # This method is meant to return a bunch of data about a user's recent courses.
-  # Right now, it just looks at recent scripts, but we expect it to change in the
-  # very near future
-  # TODO: Once this data is more real, we should be writing tests for it.
-  def recent_courses
-    in_progress_and_completed_scripts.map do |user_script|
+  # Return a collection of courses and scripts for the user. First in the list will
+  # be courses enrolled in by the user's sections. Following that will be all scripts
+  # in whcih the user has made progress that are not in any of the enrolled courses.
+  def recent_courses_and_scripts
+    courses = section_courses
+    course_scripts_script_ids = courses.map(&:course_scripts).flatten.map(&:script_id).uniq
+
+    # filter out those that are already covered by a course
+    user_scripts = in_progress_and_completed_scripts.
+      select {|user_script| !course_scripts_script_ids.include?(user_script.script_id)}
+
+    # TODO: replace courseName with name (here and in components) as some of these
+    # are courses and some are scripts
+    # TODO: may make more sense to generate links on the client (might need to
+    # provide both name and title to accomplish this). If we do we can get rid of
+    # include Rails.application.routes.url_helpers
+    # TODO: assigned sections seems like it belongs somewhere else
+    # TODO: currently no way for levelbuilders to edit description_short
```

Comment:
It might be easier for me to just own all of it, so that we don't have to worry as much about hitting merge conflicts. Shouldnt be too difficult in any case, I just didnt want to have too much change in this PR.